lib/elixir/lib/exception.ex

```diff
@@ -240,7 +240,7 @@ defmodule Exception do
   defp format_exit_reason(:noconnection), do: "no connection"
 
   defp format_exit_reason(:noproc) do
-    "no process"
+    "no process: the process is dead or no process is associated with the given name, possibly because its application isn't started"
```

Comment:
> Then it is perhaps clearer that the process having died is just one possible explanation. I do like how "died" more directly suggests one possible reason for the error, than something like "is not alive".

The reason against  "dead" or "died" is that `noproc` is about not existing - if something doesn't exist it definitely is not alive but it may never have existed so may not have died. Therefore we don't want to use dead etc.

> there is no running process

I think "running" can be used an alternative to "suspended", either at VM or `:sys` level. so would prefer to use "not alive" than running. If we want to be clear it is a moment in time: "a process is not currently associated with the given name" or similar.

> Would the above be accurate? I'm not entirely sure if you can only get :noproc when providing a name. Never when working with PIDs? Things like via tuples? Maybe they count as names?

`:global` and `:via`are global or custom naming schemes, rather than the `:local` atom names.

I would like to keep "process is not alive" at the start because that summarises the longer message. That is still true if there is no process with the name, the or just makes things more explicit.